cpp/rss_aggregator.cc

```diff
@@ -176,56 +203,72 @@ unsigned ProcessSection(const std::string &flavour, const IniFile::Section &sect
 const unsigned HARVEST_TIME_WINDOW(60); // days
 
 
-size_t SelectItems(const std::string &flavour, DbConnection * const db_connection,
+struct FeedNameAndURL {
+    std::string name_;
+    std::string url_;
+public:
+    FeedNameAndURL() = default;
+    FeedNameAndURL(const FeedNameAndURL &other) = default;
+    FeedNameAndURL(const std::string &name, const std::string &url)
+        : name_(name), url_(url) { }
+};
+
+
+size_t SelectItems(const std::string &subsystem_type, DbConnection * const db_connection,
                    std::vector<HarvestedRSSItem> * const harvested_items)
 {
-    const auto now(std::time(nullptr));
-    db_connection->queryOrDie("SELECT * FROM rss_aggregator "
-                              "WHERE pub_date >= '" + SqlUtil::TimeTToDatetime(now - HARVEST_TIME_WINDOW * 86400) + "' "
-                              "AND flavour=" + db_connection->escapeAndQuoteString(flavour) + " "
-                              "ORDER BY pub_date DESC");
-    DbResultSet result_set(db_connection->getLastResultSet());
-    while (const DbRow row = result_set.getNextRow())
-        harvested_items->emplace_back(SyndicationFormat::Item(row["item_title"], row["item_description"], row["item_url"], row["item_id"],
-                                                              SqlUtil::DatetimeToTimeT(row["pub_date"])),
-                                                              row["serial_name"], row["feed_url"]);
-    return result_set.size();
+    db_connection->queryOrDie("SELECT id,feed_name,feed_url FROM tuefind_rss_feeds WHERE FIND_IN_SET('" + subsystem_type
+                              + "', subsystem_types) > 0");
+    DbResultSet feeds_result_set(db_connection->getLastResultSet());
+    std::unordered_map<std::string, FeedNameAndURL> feed_ids_to_names_and_urls_map;
+    while (const auto row = feeds_result_set.getNextRow())
+        feed_ids_to_names_and_urls_map[row["id"]] = FeedNameAndURL(row["feed_name"], row["feed_url"]);
+
+    const std::string NOW_AS_SQL_DATETIME(SqlUtil::TimeTToDatetime(std::time(nullptr) - HARVEST_TIME_WINDOW * 86400));
```

Comment:
Incorrect variable naming => value does not equal NOW

Reply:
Done